cirq/protocols/commutes.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Protocol for determining commutativity."""
+
+from typing import Any, Union
+
+import numpy as np
+
+from cirq import linalg
+from cirq.protocols.unitary import unitary
+from cirq.type_workarounds import NotImplementedType
+
+
+def commutes(left_val: Any, right_val, atol: Union[int, float] = 1e-8) -> bool:
+    """Determines whether two values commute.
+
+    This is determined by any one of the following techniques:
+
+    - Either value has a `_commutes_` method that returns something besides
+      NotImplemented. The return value is whatever the method returned.
```

Comment:
"whatever the method returned" -> "the boolean result from this function" or something to emphasize that you can't return whatever you want in `_commutes_`

Reply:
Done